android/app/src/main/java/com/microsoft/codepush/common/CodePushStatusReportIdentifier.java

```diff
@@ -0,0 +1,116 @@
+package com.microsoft.codepush.common;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Identifier for status report saved on users device.
+ * Basically it is used in {@link com.microsoft.codepush.react.managers.CodePushTelemetryManager}
+ * for identifying status reports saved on users device. Identifier consist of two parts: deployment key and version label.
+ * If both are exist it means that update report was saved, if only version label exists it means that binary report was saved.
+ * There shouldn't be a situation when version label doesn't exist.
+ * <p>
+ * Identifier can be serialized into string for saving on device and deserialized from it.
+ */
+public class CodePushStatusReportIdentifier {
+
+    /**
+     * Deployment key.
+     */
+    private String deploymentKey;
+
+    /**
+     * Version label.
+     */
+    private String versionLabel;
+
+    /**
+     * Creates an instance of {@link CodePushStatusReportIdentifier}.
+     *
+     * @param versionLabel version label.
+     */
+    public CodePushStatusReportIdentifier(@NonNull String versionLabel) {
+        this.versionLabel = versionLabel;
+    }
+
+    /**
+     * Creates an instance of {@link CodePushStatusReportIdentifier}.
+     *
+     * @param deploymentKey deployment key.
+     * @param versionLabel  version label.
+     */
+    public CodePushStatusReportIdentifier(@NonNull String deploymentKey, @NonNull String versionLabel) {
+        this.deploymentKey = deploymentKey;
+        this.versionLabel = versionLabel;
+    }
+
+    /**
+     * Deserializes identifier from string.
+     *
+     * @param stringIdentifier input string.
+     * @return {@link CodePushStatusReportIdentifier} instance if it could be parsed, <code>null</code> otherwise.
+     */
+    public static CodePushStatusReportIdentifier fromString(String stringIdentifier) {
+        String[] parsedIdentifier = stringIdentifier.split(":");
+        if (parsedIdentifier.length > 0) {
+            return new CodePushStatusReportIdentifier(parsedIdentifier[0]);
+        } else if (parsedIdentifier.length > 1) {
+            return new CodePushStatusReportIdentifier(parsedIdentifier[0], parsedIdentifier[1]);
```

Comment:
What is `parsedIdentifier[0]` and `parsedIdentifier[1]`. It would be better to declare new descriptive variables for this array items.

Reply:
@sergey-akhalkov got it. I'll fix it.